frontend/src/metabase/visualizations/lib/apply_axis.js

```diff
@@ -243,6 +249,9 @@ export function applyChartQuantitativeXAxis(
 
     chart.xAxis().tickFormat(d => {
       // don't show ticks that aren't multiples of xInterval
+      if (waterfallTotalX === d) {
```

Comment:
It's probably fine, but I'm a bit nervous that `null` would be formatted as "Total". I doubt there's any case where a null can pop out of an axis.

Reply:
Good catch, I've added an extra safeguard!